Define a common structure for design system documentation (stories)

psatyal opened the pull request:
Creating this PR after discussion with @wielinde. The goal of this PR is to use the two "structure" stories as a base for:

- discussion
- refinement/adjustment and
- testing 

Once we are happy with a new structure for our documentation, we'll run a test by writing/re-writing the docs for one component.

https://community.openproject.org/projects/openproject/work_packages/48018/activity

psatyal committed:
Added two stories to help us better define structure

wielinde commented on frontend/src/stories/Structure-DatePicker.mdx:
```diff
@@ -0,0 +1,80 @@
+- Individual date selection not possible via keyboard navigation, known Flatpickr issue
+	- If selectable, label ("13") is not enough; screen reader should read the full date, "Friday, 13 October, 2023")
+
+## Angular components
```
@psatyal As I understand Storybook now, I think that every Angular Component should have its own mdx file. And each variations of an Angular component shall have their own story. In other words, I don't think that it will work to have all Angular components listed in one mdx file as sketched here. @b12f correct?

b12f replied:
You can have as many `mdx` files as you like; they're just markdown files. It also sometimes makes sense to explain multiple angular components in one place. For example, both the single date picker and range date picker have their own components, but it makes sense to document them and show their stories in the same mdx.

wielinde replied:
@b12f Oh, I see. So the (generated) documentation for inputs and outputs (the "Controls" tab) is per story and independent from mdx file?

b12f replied:
The documentation for inputs and outputs is generated by using the `ArgsTable` react component inside of an `mdx` file and giving it a reference to the angular component class it should be showing the args for: https://github.com/opf/openproject/blob/dev/frontend/src/app/spot/components/checkbox/stories/Checkbox.mdx?plain=1#L18

wielinde reviewed (commented):
Thanks @psatyal. Please see my little comment.

psatyal committed:
Updates to the structure and example stories

psatyal committed:
Added "Structure" page, updated Datepicker and Breadcrumbs

psatyal commented:
Hi team, 

I've made these changes:

- Added a "Story Structure" page to the top with an outline of the structure we discussed on Thursday
- Updated the Datepicker and Breadcrumbs stories based on this structure

I've added @marcalcobe, @b12f and @wielinde for review. We can discuss it on Monday, but if OK on your guys' end, this can close this task: https://community.openproject.org/work_packages/48018/activity

marcalcobe reviewed (commented):
Hi @psatyal,
I've reviewed the general structure proposal and I find it good. The only question I have is if there is a component that each variation might have different behaviours, I assume the shared/global behaviours will go to the "Behaviour" and then the behavioural variations inside of each variation/type inside of the "Variations" section?

psatyal commented on frontend/src/stories/Breadcrumbs.mdx:
```diff
@@ -15,8 +15,6 @@ Breadcrumbs allow users to:
 
-## Structure and Options
-
 Breadcrumbs consist of a series of "levels" presented horizontally, separated by a right arrow (▶). Each level is a child of the level preceding it, except the first level, which is the root.
```
Here I can leave a comment. (Just a test)

as-op commented:
Hi @psatyal, I'm closing this storybook PR as it has conflicts. And as we moved to primer, I think it's outdated.
